RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitProducerGPU.cc

```diff
@@ -111,7 +111,17 @@ EcalUncalibRecHitProducerGPU::EcalUncalibRecHitProducerGPU(const edm::ParameterS
     : digisTokenEB_{consumes<InputProduct>(ps.getParameter<edm::InputTag>("digisLabelEB"))},
       digisTokenEE_{consumes<InputProduct>(ps.getParameter<edm::InputTag>("digisLabelEE"))},
       recHitsTokenEB_{produces<OutputProduct>(ps.getParameter<std::string>("recHitsLabelEB"))},
-      recHitsTokenEE_{produces<OutputProduct>(ps.getParameter<std::string>("recHitsLabelEE"))} {
+      recHitsTokenEE_{produces<OutputProduct>(ps.getParameter<std::string>("recHitsLabelEE"))},
+      pedestalsToken_{esConsumes<EcalPedestalsGPU, EcalPedestalsRcd>()},
+      gainRatiosToken_{esConsumes<EcalGainRatiosGPU, EcalGainRatiosRcd>()},
+      pulseShapesToken_{esConsumes<EcalPulseShapesGPU, EcalPulseShapesRcd>()},
+      pulseCovariancesToken_{esConsumes<EcalPulseCovariancesGPU, EcalPulseCovariancesRcd>()},
+      samplesCorrelationToken_{esConsumes<EcalSamplesCorrelationGPU, EcalSamplesCorrelationRcd>()},
+      timeBiasCorrectionsToken_{esConsumes<EcalTimeBiasCorrectionsGPU, EcalTimeBiasCorrectionsRcd>()},
+      timeCalibConstantsToken_{esConsumes<EcalTimeCalibConstantsGPU, EcalTimeCalibConstantsRcd>()},
+      sampleMaskToken_{esConsumes<EcalSampleMask, EcalSampleMaskRcd>()},
+      timeOffsetConstantToken_{esConsumes<EcalTimeOffsetConstant, EcalTimeOffsetConstantRcd>()},
+      multifitParametersToken_{esConsumes<EcalMultifitParametersGPU, EcalMultifitParametersGPURecord>()} {
```

Comment:
@makortel a general comment abut `ESGetToken`: it looks to me like an `EDGetToken` object already knows the types that need to be passed to `esConsumes<...>()`.

Why not add a simpler `ESGetToken` constructor that takes as argument only a pointer or reference to the module being constructed ?
So that a like like
```c++
      multifitParametersToken_{esConsumes<EcalMultifitParametersGPU, EcalMultifitParametersGPURecord>()},
```
could become
```c++
      multifitParametersToken_(*this)),
```
and the constructor can take care of calling `module.esConsumes<T, R>()` ?

Reply:
I suppose we could add something to avoid repeating the types (similar to what is available in ESProducers). Could you open an issue about it?

Reply:
https://github.com/cms-sw/cmssw/issues/30828